pkg/v1/remote/catalog.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
+)
+
+type catalog struct {
+	Repos []string `json:"repositories"`
+}
+
+// GetCatalog calls /_catalog, returning the list of repositories on the registry
+func GetCatalog(target name.Registry, options ...Option) ([]string, error) {
```

Comment:
Given that the results of catalog can be paginated, there might be a better return type than `[]string` -- I'm not sure what the most go idiomatic thing to do here is...

We could return a channel? Or do callbacks for each "page"?

Or maybe this is just fine...

Reply:
what about now? (sorry to double comment)

Reply:
I think it'd make sense for `GetCatalog` to do pagination itself, and return all the results it could find from however many requests were necessary to find them. That seems simpler than making callers deal with pagination, or pass a channel, or whatever.